**DELETE & CREATE SOLR COLLECTION:**
```
## delete and create core
/local/software/solr-6.2.0/bin/solr delete -c seqrdb && \
/local/software/solr-6.2.0/bin/solr create_collection -c seqrdb -shards 3
```


```
## import data
INMR_v9: read -i file:///home/users/weisburd/data/projects/test/wes/INMR_v9/INMR_v9.vds
Hanson:  read -i file:///home/users/weisburd/data/projects/test/wgs/Hanson-gatk3dot4/Hanson.vds 
NIAMS:   read -i file:///home/users/weisburd/data/projects/test/wgs/NIAMS-gatk3dot4/NIAMS.vds 
NIAID:   read -i file:///home/users/weisburd/data/projects/test/wgs/NIAID-gatk3dot4/NIAID.vds 
```

**HAIL COMMAND:**

```
time hail_local_with_3_cores \
  read -i file:///home/users/weisburd/data/projects/test/wes/INMR_v9/INMR_v9.vds \
  annotatevariants vds -r va.clinvar -i file:///mnt/lustre/weisburd/data/reference_data/clinvar/clinvar_v2016_09_01.vds \
  annotatevariants vds -r va.exac -i file:///mnt/lustre/weisburd/data/reference_data/exac/exac_v1.vds \
  annotatevariants vds -r va.g1k -i file:///mnt/lustre/weisburd/data/reference_data/1kg/1kg_wgs_phase3.vds \
  annotatevariants vds -r va.dbnsfp -i file:///mnt/lustre/weisburd/data/reference_data/dbnsfp/dbNSFP_3.2a_variant.filtered.allhg19_nodup.vds \
  annotatevariants expr -c 'va.vep.sorted_transcript_consequences = va.vep.transcript_consequences.map(c => select(c, amino_acids, biotype, canonical, cdna_start, cdna_end, codons, consequence_terms, distance, domains, exon, gene_id, transcript_id, protein_id, gene_symbol, gene_symbol_source, hgnc_id, hgvsc, hgvsp, lof, lof_flags, lof_filter, lof_info)).sortBy(c => 
        let is_coding=(c.biotype=="protein_coding") and is_most_severe=c.consequence_terms.toSet.contains(va.vep.most_severe_consequence) and is_canonical=(c.canonical==1) in
            if(is_coding) 
                if(is_most_severe) 
                    if(is_canonical)  1  else  2
                else  3
            else
                if(is_most_severe) 
                    if(is_canonical)  4  else  5
                else  6
        )' \
  exportvariantssolr --export-ref -c INMR_v9_with_ref -v '  
    dataset_id = "INMR_v9",
    dataset_version = "2016_04_12",
    dataset_type = "wex",
    chrom = v.contig,
    start = v.start,
    end = va.info.END,
    ref = v.ref,
    alt = v.alt,
    filters = va.filters,
    pass = va.pass,
    rsid = va.rsid,
    AC = va.info.AC[va.aIndex-1],
    AN = va.info.AN,
    AF = va.info.AF[va.aIndex-1],
    was_split = va.wasSplit,
    clinvar_clinsig = va.clinvar.clinical_significance,
    clinvar_review_status = va.clinvar.review_status,
    clinvar_is_pathogenic = va.clinvar.pathogenic,
    clinvar_is_conflicted = va.clinvar.conflicted,
    dbnsfp_sift_pred { indexed = false } = va.dbnsfp.SIFT_pred, 
    dbnsfp_polyphen2_hdiv_pred { indexed = false } = va.dbnsfp.Polyphen2_HDIV_pred,
    dbnsfp_polyphen2_hvar_pred { indexed = false } = va.dbnsfp.Polyphen2_HVAR_pred,
    dbnsfp_lrt_pred { indexed = false } = va.dbnsfp.LRT_pred,
    dbnsfp_muttaster_pred { indexed = false } = va.dbnsfp.MutationTaster_pred,
    dbnsfp_mutassesor_pred { indexed = false } = va.dbnsfp.MutationAssessor_pred,
    dbnsfp_fathmm_pred { indexed = false } = va.dbnsfp.FATHMM_pred,
    dbnsfp_provean_pred { indexed = false } = va.dbnsfp.PROVEAN_pred,
    dbnsfp_metasvm_pred { indexed = false } = va.dbnsfp.MetaSVM_pred,
    dbnsfp_metalr_pred { indexed = false } = va.dbnsfp.MetaLR_pred,
    dbnsfp_cadd_phred { indexed = false } = va.dbnsfp.CADD_phred,
    vep_consequences = va.vep.transcript_consequences.map( x => x.consequence_terms ).flatten().toSet,
    vep_gene_ids = va.vep.transcript_consequences.map( x => x.gene_id ).toSet,
    vep_transcript_ids = va.vep.transcript_consequences.map( x => x.transcript_id ).toSet,
    vep_most_severe_consequence = va.vep.most_severe_consequence,
    vep_annotations_sorted { type = "text_ws", indexed = false } = json(va.vep.sorted_transcript_consequences),
    g1k_wgs_phase3_afr_af = va.g1k.info.AFR_AF[va.g1k.aIndex-1],
    g1k_wgs_phase3_amr_af = va.g1k.info.AMR_AF[va.g1k.aIndex-1],
    g1k_wgs_phase3_eur_af = va.g1k.info.EUR_AF[va.g1k.aIndex-1],
    g1k_wgs_phase3_eas_af = va.g1k.info.EAS_AF[va.g1k.aIndex-1],
    g1k_wgs_phase3_sas_af = va.g1k.info.SAS_AF[va.g1k.aIndex-1],
    g1k_wgs_phase3_global_af = va.g1k.info.AF[va.g1k.aIndex-1],
    g1k_wgs_phase3_popmax_af = va.g1k.info.POPMAX_AF,
    g1k_wgs_phase3_popmax = va.g1k.info.POPMAX,
    exac_v1_afr_af = if(va.exac.info.AN_AFR == 0) NA:Double else va.exac.info.AC_AFR[va.exac.aIndex-1]/va.exac.info.AN_AFR,
    exac_v1_amr_af = if(va.exac.info.AN_AMR == 0) NA:Double else va.exac.info.AC_AMR[va.exac.aIndex-1]/va.exac.info.AN_AMR,
    exac_v1_nfe_af = if(va.exac.info.AN_NFE == 0) NA:Double else va.exac.info.AC_NFE[va.exac.aIndex-1]/va.exac.info.AN_NFE,
    exac_v1_fin_af = if(va.exac.info.AN_FIN == 0) NA:Double else va.exac.info.AC_FIN[va.exac.aIndex-1]/va.exac.info.AN_FIN,
    exac_v1_eas_af = if(va.exac.info.AN_EAS == 0) NA:Double else va.exac.info.AC_EAS[va.exac.aIndex-1]/va.exac.info.AN_EAS,
    exac_v1_sas_af = if(va.exac.info.AN_SAS == 0) NA:Double else va.exac.info.AC_SAS[va.exac.aIndex-1]/va.exac.info.AN_SAS,
    exac_v1_global_af = va.exac.info.AF[va.exac.aIndex-1],
    exac_v1_popmax_af = if(va.exac.info.AN_POPMAX[va.exac.aIndex-1] == 0) NA:Double else va.exac.info.AC_POPMAX[va.exac.aIndex-1]/va.exac.info.AN_POPMAX[va.exac.aIndex-1],
    exac_v1_popmax = va.exac.info.POPMAX[va.exac.aIndex-1],
    twinsuk_af = va.dbnsfp.TWINSUK_AF.toDouble,
    alspac_af = va.dbnsfp.ALSPAC_AF.toDouble,
    esp65000_aa_af = va.dbnsfp.ESP6500_AA_AF.toDouble,
    esp65000_ea_af = va.dbnsfp.ESP6500_EA_AF.toDouble
    ' \
  -g 'num_alt = g.nNonRefAlleles,
    gq = g.gq,
    ab = let s = g.ad.sum in if (s == 0) NA:Float else (g.ad[0] / s).toFloat, 
    dp = g.dp,
    pl { indexed = false } = g.pl' \
  -z 'seqr-db1:2181,seqr-db2:2181,seqr-db3:2181'
```


**Available Reference Annotations:**  

```
hail: info: running: printschema
Global annotation schema:
global: Empty

Sample annotation schema:
sa: Empty

Variant annotation schema:
va: Struct {
    rsid: String,
    qual: Double,
    filters: Set[String],
    pass: Boolean,
    info: Struct {
        AC: Array[Int],
        AF: Array[Double],
        AN: Int,
        BaseQRankSum: Double,
        CCC: Int,
        ClippingRankSum: Double,
        DB: Boolean,
        DP: Int,
        DS: Boolean,
        END: Int,
        FS: Double,
        GQ_MEAN: Double,
        GQ_STDDEV: Double,
        HWP: Double,
        HaplotypeScore: Double,
        InbreedingCoeff: Double,
        MLEAC: Array[Int],
        MLEAF: Array[Double],
        MQ: Double,
        MQ0: Int,
        MQRankSum: Double,
        NCC: Int,
        NEGATIVE_TRAIN_SITE: Boolean,
        POSITIVE_TRAIN_SITE: Boolean,
        QD: Double,
        ReadPosRankSum: Double,
        VQSLOD: Double,
        culprit: String
    },
    aIndex: Int,
    wasSplit: Boolean,
    vep: Struct {
        assembly_name: String,
        allele_string: String,
        ancestral: String,
        colocated_variants: Array[Struct {
            aa_allele: String,
            aa_maf: Double,
            afr_allele: String,
            afr_maf: Double,
            allele_string: String,
            amr_allele: String,
            amr_maf: Double,
            clin_sig: Array[String],
            end: Int,
            eas_allele: String,
            eas_maf: Double,
            ea_allele: String,
            ea_maf: Double,
            eur_allele: String,
            eur_maf: Double,
            exac_adj_allele: String,
            exac_adj_maf: Double,
            exac_allele: String,
            exac_afr_allele: String,
            exac_afr_maf: Double,
            exac_amr_allele: String,
            exac_amr_maf: Double,
            exac_eas_allele: String,
            exac_eas_maf: Double,
            exac_fin_allele: String,
            exac_fin_maf: Double,
            exac_maf: Double,
            exac_nfe_allele: String,
            exac_nfe_maf: Double,
            exac_oth_allele: String,
            exac_oth_maf: Double,
            exac_sas_allele: String,
            exac_sas_maf: Double,
            id: String,
            minor_allele: String,
            minor_allele_freq: Double,
            phenotype_or_disease: Int,
            pubmed: Array[Int],
            sas_allele: String,
            sas_maf: Double,
            somatic: Int,
            start: Int,
            strand: Int
        }],
        context: String,
        end: Int,
        id: String,
        input: String,
        intergenic_consequences: Array[Struct {
            allele_num: Int,
            consequence_terms: Array[String],
            impact: String,
            minimised: Int,
            variant_allele: String
        }],
        most_severe_consequence: String,
        motif_feature_consequences: Array[Struct {
            allele_num: Int,
            consequence_terms: Array[String],
            high_inf_pos: String,
            impact: String,
            minimised: Int,
            motif_feature_id: String,
            motif_name: String,
            motif_pos: Int,
            motif_score_change: Double,
            strand: Int,
            variant_allele: String
        }],
        regulatory_feature_consequences: Array[Struct {
            allele_num: Int,
            biotype: String,
            consequence_terms: Array[String],
            impact: String,
            minimised: Int,
            regulatory_feature_id: String,
            variant_allele: String
        }],
        seq_region_name: String,
        start: Int,
        strand: Int,
        transcript_consequences: Array[Struct {
            allele_num: Int,
            amino_acids: String,
            biotype: String,
            canonical: Int,
            ccds: String,
            cdna_start: Int,
            cdna_end: Int,
            cds_end: Int,
            cds_start: Int,
            codons: String,
            consequence_terms: Array[String],
            distance: Int,
            domains: Array[Struct {
                db: String,
                name: String
            }],
            exon: String,
            gene_id: String,
            gene_pheno: Int,
            gene_symbol: String,
            gene_symbol_source: String,
            hgnc_id: Int,
            hgvsc: String,
            hgvsp: String,
            hgvs_offset: Int,
            impact: String,
            intron: String,
            lof: String,
            lof_flags: String,
            lof_filter: String,
            lof_info: String,
            minimised: Int,
            polyphen_prediction: String,
            polyphen_score: Double,
            protein_end: Int,
            protein_start: Int,
            protein_id: String,
            sift_prediction: String,
            sift_score: Double,
            strand: Int,
            swissprot: String,
            transcript_id: String,
            trembl: String,
            uniparc: String,
            variant_allele: String
        }],
        variant_class: String
    },
    g1k: Struct {
        rsid: String,
        qual: Double,
        filters: Set[String],
        pass: Boolean,
        info: Struct {
            CIEND: Array[Int],
            CIPOS: Array[Int],
            CS: String,
            END: Int,
            IMPRECISE: Boolean,
            MC: Array[String],
            MEINFO: Array[String],
            MEND: Int,
            MLEN: Int,
            MSTART: Int,
            SVLEN: Array[Int],
            SVTYPE: String,
            TSD: String,
            AC: Array[Int],
            AF: Array[Double],
            NS: Int,
            AN: Int,
            EAS_AF: Array[Double],
            EUR_AF: Array[Double],
            AFR_AF: Array[Double],
            AMR_AF: Array[Double],
            SAS_AF: Array[Double],
            DP: Int,
            AA: String,
            VT: Array[String],
            EX_TARGET: Boolean,
            MULTI_ALLELIC: Boolean,
            POPMAX_AF: Double,
            POPMAX: String
        },
        aIndex: Int,
        wasSplit: Boolean
    },
    exac: Struct {
        rsid: String,
        qual: Double,
        filters: Set[String],
        pass: Boolean,
        info: Struct {
            AC: Array[Int],
            AC_AFR: Array[Int],
            AC_AMR: Array[Int],
            AC_Adj: Array[Int],
            AC_EAS: Array[Int],
            AC_FIN: Array[Int],
            AC_Hemi: Array[Int],
            AC_Het: Array[Int],
            AC_Hom: Array[Int],
            AC_NFE: Array[Int],
            AC_OTH: Array[Int],
            AC_SAS: Array[Int],
            AF: Array[Double],
            AN: Int,
            AN_AFR: Int,
            AN_AMR: Int,
            AN_Adj: Int,
            AN_EAS: Int,
            AN_FIN: Int,
            AN_NFE: Int,
            AN_OTH: Int,
            AN_SAS: Int,
            BaseQRankSum: Double,
            CCC: Int,
            ClippingRankSum: Double,
            DB: Boolean,
            DP: Int,
            DS: Boolean,
            END: Int,
            FS: Double,
            GQ_MEAN: Double,
            GQ_STDDEV: Double,
            HWP: Double,
            HaplotypeScore: Double,
            Hemi_AFR: Array[Int],
            Hemi_AMR: Array[Int],
            Hemi_EAS: Array[Int],
            Hemi_FIN: Array[Int],
            Hemi_NFE: Array[Int],
            Hemi_OTH: Array[Int],
            Hemi_SAS: Array[Int],
            Het_AFR: Array[Int],
            Het_AMR: Array[Int],
            Het_EAS: Array[Int],
            Het_FIN: Array[Int],
            Het_NFE: Array[Int],
            Het_OTH: Array[Int],
            Het_SAS: Array[Int],
            Hom_AFR: Array[Int],
            Hom_AMR: Array[Int],
            Hom_EAS: Array[Int],
            Hom_FIN: Array[Int],
            Hom_NFE: Array[Int],
            Hom_OTH: Array[Int],
            Hom_SAS: Array[Int],
            InbreedingCoeff: Double,
            MLEAC: Array[Int],
            MLEAF: Array[Double],
            MQ: Double,
            MQ0: Int,
            MQRankSum: Double,
            NCC: Int,
            NEGATIVE_TRAIN_SITE: Boolean,
            POSITIVE_TRAIN_SITE: Boolean,
            QD: Double,
            ReadPosRankSum: Double,
            VQSLOD: Double,
            culprit: String,
            DP_HIST: Array[String],
            GQ_HIST: Array[String],
            DOUBLETON_DIST: Array[String],
            AC_MALE: Array[String],
            AC_FEMALE: Array[String],
            AN_MALE: String,
            AN_FEMALE: String,
            AC_CONSANGUINEOUS: Array[String],
            AN_CONSANGUINEOUS: String,
            Hom_CONSANGUINEOUS: Array[String],
            CSQ: Boolean,
            AC_POPMAX: Array[String],
            AN_POPMAX: Array[String],
            POPMAX: Array[String],
            clinvar_measureset_id: Array[String],
            clinvar_conflicted: Array[String],
            clinvar_pathogenic: Array[String],
            clinvar_mut: Array[String],
            K1_RUN: Array[String],
            K2_RUN: Array[String],
            K3_RUN: Array[String],
            ESP_AF_POPMAX: Array[String],
            ESP_AF_GLOBAL: Array[String],
            ESP_AC: Array[String],
            KG_AF_POPMAX: Array[String],
            KG_AF_GLOBAL: Array[String],
            KG_AC: Array[String]
        },
        aIndex: Int,
        wasSplit: Boolean
    },
    clinvar: Struct {
        chrom: String,
        pos: Int,
        ref: String,
        alt: String,
        mut: String,
        measureset_id: Int,
        symbol: String,
        clinical_significance: String,
        review_status: String,
        hgvs_c: String,
        hgvs_p: String,
        all_submitters: String,
        all_traits: String,
        all_pmids: String,
        pathogenic: Int,
        conflicted: Int
    },
    dbnsfp: Struct {
        `#chr`: String,
        `pos(1-based)`: String,
        ref: String,
        alt: String,
        aaref: String,
        aaalt: String,
        rs_dbSNP146: String,
        hg19_chr: String,
        `hg19_pos(1-based)`: Int,
        hg18_chr: String,
        `hg18_pos(1-based)`: String,
        genename: String,
        cds_strand: String,
        refcodon: String,
        codonpos: String,
        codon_degeneracy: String,
        Ancestral_allele: String,
        AltaiNeandertal: String,
        Denisova: String,
        Ensembl_geneid: String,
        Ensembl_transcriptid: String,
        Ensembl_proteinid: String,
        aapos: String,
        SIFT_score: String,
        SIFT_converted_rankscore: String,
        SIFT_pred: String,
        Uniprot_acc_Polyphen2: String,
        Uniprot_id_Polyphen2: String,
        Uniprot_aapos_Polyphen2: String,
        Polyphen2_HDIV_score: String,
        Polyphen2_HDIV_rankscore: String,
        Polyphen2_HDIV_pred: String,
        Polyphen2_HVAR_score: String,
        Polyphen2_HVAR_rankscore: String,
        Polyphen2_HVAR_pred: String,
        LRT_score: String,
        LRT_converted_rankscore: String,
        LRT_pred: String,
        LRT_Omega: String,
        MutationTaster_score: String,
        MutationTaster_converted_rankscore: String,
        MutationTaster_pred: String,
        MutationTaster_model: String,
        MutationTaster_AAE: String,
        MutationAssessor_UniprotID: String,
        MutationAssessor_variant: String,
        MutationAssessor_score: String,
        MutationAssessor_score_rankscore: String,
        MutationAssessor_pred: String,
        FATHMM_score: String,
        FATHMM_converted_rankscore: String,
        FATHMM_pred: String,
        PROVEAN_score: String,
        PROVEAN_converted_rankscore: String,
        PROVEAN_pred: String,
        Transcript_id_VEST3: String,
        Transcript_var_VEST3: String,
        VEST3_score: String,
        VEST3_rankscore: String,
        MetaSVM_score: String,
        MetaSVM_rankscore: String,
        MetaSVM_pred: String,
        MetaLR_score: String,
        MetaLR_rankscore: String,
        MetaLR_pred: String,
        Reliability_index: String,
        CADD_raw: String,
        CADD_raw_rankscore: String,
        CADD_phred: String,
        DANN_score: String,
        DANN_rankscore: String,
        `fathmm-MKL_coding_score`: String,
        `fathmm-MKL_coding_rankscore`: String,
        `fathmm-MKL_coding_pred`: String,
        `fathmm-MKL_coding_group`: String,
        `Eigen-raw`: String,
        `Eigen-phred`: String,
        `Eigen-raw_rankscore`: String,
        `Eigen-PC-raw`: String,
        `Eigen-PC-raw_rankscore`: String,
        GenoCanyon_score: String,
        GenoCanyon_score_rankscore: String,
        integrated_fitCons_score: String,
        integrated_fitCons_score_rankscore: String,
        integrated_confidence_value: String,
        GM12878_fitCons_score: String,
        GM12878_fitCons_score_rankscore: String,
        GM12878_confidence_value: String,
        `H1-hESC_fitCons_score`: String,
        `H1-hESC_fitCons_score_rankscore`: String,
        `H1-hESC_confidence_value`: String,
        HUVEC_fitCons_score: String,
        HUVEC_fitCons_score_rankscore: String,
        HUVEC_confidence_value: String,
        `GERP++_NR`: String,
        `GERP++_RS`: String,
        `GERP++_RS_rankscore`: String,
        phyloP100way_vertebrate: String,
        phyloP100way_vertebrate_rankscore: String,
        phyloP20way_mammalian: String,
        phyloP20way_mammalian_rankscore: String,
        phastCons100way_vertebrate: String,
        phastCons100way_vertebrate_rankscore: String,
        phastCons20way_mammalian: String,
        phastCons20way_mammalian_rankscore: String,
        SiPhy_29way_pi: String,
        SiPhy_29way_logOdds: String,
        SiPhy_29way_logOdds_rankscore: String,
        `1000Gp3_AC`: String,
        `1000Gp3_AF`: String,
        `1000Gp3_AFR_AC`: String,
        `1000Gp3_AFR_AF`: String,
        `1000Gp3_EUR_AC`: String,
        `1000Gp3_EUR_AF`: String,
        `1000Gp3_AMR_AC`: String,
        `1000Gp3_AMR_AF`: String,
        `1000Gp3_EAS_AC`: String,
        `1000Gp3_EAS_AF`: String,
        `1000Gp3_SAS_AC`: String,
        `1000Gp3_SAS_AF`: String,
        TWINSUK_AC: String,
        TWINSUK_AF: String,
        ALSPAC_AC: String,
        ALSPAC_AF: String,
        ESP6500_AA_AC: String,
        ESP6500_AA_AF: String,
        ESP6500_EA_AC: String,
        ESP6500_EA_AF: String,
        ExAC_AC: String,
        ExAC_AF: String,
        ExAC_Adj_AC: String,
        ExAC_Adj_AF: String,
        ExAC_AFR_AC: String,
        ExAC_AFR_AF: String,
        ExAC_AMR_AC: String,
        ExAC_AMR_AF: String,
        ExAC_EAS_AC: String,
        ExAC_EAS_AF: String,
        ExAC_FIN_AC: String,
        ExAC_FIN_AF: String,
        ExAC_NFE_AC: String,
        ExAC_NFE_AF: String,
        ExAC_SAS_AC: String,
        ExAC_SAS_AF: String,
        ExAC_nonTCGA_AC: String,
        ExAC_nonTCGA_AF: String,
        ExAC_nonTCGA_Adj_AC: String,
        ExAC_nonTCGA_Adj_AF: String,
        ExAC_nonTCGA_AFR_AC: String,
        ExAC_nonTCGA_AFR_AF: String,
        ExAC_nonTCGA_AMR_AC: String,
        ExAC_nonTCGA_AMR_AF: String,
        ExAC_nonTCGA_EAS_AC: String,
        ExAC_nonTCGA_EAS_AF: String,
        ExAC_nonTCGA_FIN_AC: String,
        ExAC_nonTCGA_FIN_AF: String,
        ExAC_nonTCGA_NFE_AC: String,
        ExAC_nonTCGA_NFE_AF: String,
        ExAC_nonTCGA_SAS_AC: String,
        ExAC_nonTCGA_SAS_AF: String,
        ExAC_nonpsych_AC: String,
        ExAC_nonpsych_AF: String,
        ExAC_nonpsych_Adj_AC: String,
        ExAC_nonpsych_Adj_AF: String,
        ExAC_nonpsych_AFR_AC: String,
        ExAC_nonpsych_AFR_AF: String,
        ExAC_nonpsych_AMR_AC: String,
        ExAC_nonpsych_AMR_AF: String,
        ExAC_nonpsych_EAS_AC: String,
        ExAC_nonpsych_EAS_AF: String,
        ExAC_nonpsych_FIN_AC: String,
        ExAC_nonpsych_FIN_AF: String,
        ExAC_nonpsych_NFE_AC: String,
        ExAC_nonpsych_NFE_AF: String,
        ExAC_nonpsych_SAS_AC: String,
        ExAC_nonpsych_SAS_AF: String,
        clinvar_rs: String,
        clinvar_clnsig: String,
        clinvar_trait: String,
        clinvar_golden_stars: String,
        Interpro_domain: String,
        GTEx_V6_gene: String,
        GTEx_V6_tissue: String
    }
}
hail: info: timing:
  read: 8.084s
  annotatevariants vds: 2.467s
  annotatevariants vds: 585.568ms
  annotatevariants vds: 2.076s
  annotatevariants vds: 1.505s
  printschema: 12.472ms

real	0m21.911s
user	0m20.130s
sys	0m1.134s

```
